src/core/mac/sub_mac.cpp

```diff
@@ -247,7 +247,7 @@ Error SubMac::Sleep(void)
 
     VerifyOrExit(ShouldHandleTransitionToSleep());
 
-    error = Get<Radio>().Sleep();
+    SuccessOrExit(error = RadioSleep());
```

Comment:
One thing related to radio state and `OT_RADIO_CAPS_RX_ON_WHEN_IDLE`: if this capability is used, then `SubMac::Sleep()` will never ask the radio to sleep directly (skips calling `RadioSleep()`). So, the `RadioScheduler` won't be notified of this.

The addition of the `OT_RADIO_CAPS_RX_ON_WHEN_IDLE` feature makes this whole change (adding `RadioScheduler`) to manage sleep/receive more complicated.

Reply:
I have updated the code to process `OT_RADIO_CAPS_RX_ON_WHEN_IDLE`.